jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -609,29 +609,22 @@ ecma_string_get_array_index (const ecma_string_t *str_p, /**< ecma-string */
 
 /**
  * Convert ecma-string's contents to a cesu-8 string and put it to the buffer.
+ * It is the caller's responsibility to make sure that the string fits in the buffer.
  *
- * @return number of bytes, actually copied to the buffer - if string's content was copied successfully;
- *         otherwise (in case size of buffer is insufficient) - negative number, which is calculated
- *         as negation of buffer size, that is required to hold the string's content.
+ * @return number of bytes, actually copied to the buffer.
```

Comment:
Well, we fail with an assert (in debug). That's aligned with other similar functions (e.g., `ecma_uint32_to_utf8_string`).

Actually, at almost all call sites, we rely on the buffer being preallocated to the exact size of the string. Perhaps the only counter example is `ecma_compare_ecma_strings_relational`, where a buffer of fixed size is available and there was a logic to decide whether to used that fixed buffer or allocate a new one, and that logic really relied on a negative return value from `ecma_string_to_utf8_string`, but it was to rewrite to use `ecma_string_get_size ` instead.